### ISSUE TITLE
Allow plugin directories to be optional

### ISSUE THREAD

**Issue by skellock** [infinitered/gluegun#248]
Using `gluegun@2.0.0-alpha.2`, I have this in my runtime config:

```ts
    .plugin(resolve(`${filesystem.cwd()}`, BRAND), LOAD_OPTIONS)
```

But I want that to be optional as it may not exist.

Here's the error it spits out:

```
Error: Error: couldn't load plugin (not a directory): /Users/steve/src/ir/app-machine/app-machine
    at loadFromDirectory (/Users/steve/src/ir/app-machine/node_modules/gluegun/src/loaders/toml-plugin-loader.js:40:11)
    at Runtime.load (/Users/steve/src/ir/app-machine/node_modules/gluegun/src/domain/runtime.js:184:20)
    at loadPlugins.forEach.entry (/Users/steve/src/ir/app-machine/node_modules/gluegun/src/domain/builder.js:56:19)
    at Array.forEach (<anonymous>)
    at Builder.create (/Users/steve/src/ir/app-machine/node_modules/gluegun/src/domain/builder.js:50:22)
    at Object.configureRuntime (/Users/steve/src/ir/app-machine/src/cli/gluegun-runtime.ts:20:6)
    at Object.<anonymous> (/Users/steve/src/ir/app-machine/src/cli/cli.ts:24:21)
    at step (/Users/steve/src/ir/app-machine/src/cli/cli.ts:32:23)
    at Object.next (/Users/steve/src/ir/app-machine/src/cli/cli.ts:13:53)
    at /Users/steve/src/ir/app-machine/src/cli/cli.ts:7:71
```

For `src` I can see why it's required, but for `plugin` and `plugins`, it should be optional.

**Comment by skellock** [comment]
I think we need something before this check:

https://github.com/infinitered/gluegun/blob/1cf8014eff543d8996f41a09b72d8f9829576242/packages/gluegun/src/loaders/toml-plugin-loader.js#L38-L43

If it doesn't exist, we should return gracefully.

**Comment by skellock** [comment]
Perhaps a better option would be in the runtime.

https://github.com/infinitered/gluegun/blob/1cf8014eff543d8996f41a09b72d8f9829576242/packages/gluegun/src/domain/runtime.js#L178-L194

This way a `null` won't get added to the plugins list.

**Comment by WilliamStephens** [comment]
Hey, I would love to take this one!

**Comment by skellock** [comment]
Sure!  I was thinking  of `options` having an extra property called `required` which wouldn't be there (so falsy) by default.  But in `loadDefault`, it would be `required: true`.

I think.  Haha.

**Comment by jamonholmgren** [comment]
@WilliamStephens Let me know if you need any help, would be happy to give pointers!

**Comment by WilliamStephens** [comment]
@jamonholmgren My bad forgot about this. I can probably get to it later today. I was thinking about just adding required to the options and then setting a flag on the plugin maybe something like isValid. Then in the runtime only add plugins that are valid. I'm assuming it should still throw if required is true and the path is not valid? I was also wondering if you guys had any kind of logger to let the user know that the plugin wasn't loaded.

**Comment by jamonholmgren** [comment]
> I'm assuming it should still throw if required is true and the path is not valid?

Yes.

> I was also wondering if you guys had any kind of logger to let the user know that the plugin wasn't loaded.

@skellock says Gluegun shouldn't print anything, so I'm assuming not. Maybe we need to allow them to provide a callback function in case a plugin isn't valid, to allow them to print that?

**Comment by skellock** [comment]
I'm cool with blowing up if the plugin is invalid.  

We used to catch that error and tuck it away in safe spot for reporting.  Nobody seemed to like that approach so we ditched it in v2.

This though is different I think.  This is just:  ya, that plugin dir.  It's not even there, so I'm not going to try to load it.

`loadAll` does something similar.  `load` doesn't.   `loadDefault` should definitely blow up if it's not found, but `load` should be ok to proceed without it.
